source/MaterialXRuntime/RtApi.cpp

```diff
@@ -229,6 +279,43 @@ const FileSearchPath& RtApi::getImplementationSearchPath() const
     return _cast(_ptr)->getImplementationSearchPath();
 }
 
+const FileSearchPath RtApi::getResolvedImagePaths(bool includeSubFolders) const
+{
+    RtStagePtr stage = getStage(0);
+    // Image search path building.
+    FileSearchPath imageSearchPath;
+    for (const FilePath& uri : stage->getSourceUri())
+    {
+        FilePath path(uri);
+        if (stringEndsWith(uri.asString(), ".mtlx"))
+        {
+            path = path.getParentPath();
+        }
+        imageSearchPath.append(path);
+    }
+    FileSearchPath searchPath = getTextureSearchPath();
+    FilePathVec libraryFolders;
+    for (size_t i=0; i<searchPath.size(); i++)
+    {
+        libraryFolders.push_back(searchPath[i]);
+    }
+    if (includeSubFolders)
+    {
+        FilePathVec childFolders;
+        getSubdirectories(libraryFolders, searchPath, childFolders);
+        libraryFolders.insert(std::end(libraryFolders), std::begin(childFolders), std::end(childFolders));
+        for (const auto& childFolder : childFolders)
+        {
+            imageSearchPath.append(childFolder);
+        }
+    }
```

Comment:
We should add the definition paths here, also if inludeSubFolders is false, nothing seems to be updated.

Reply:
Ok, so append the definition paths to the imageSearchPath, will do.